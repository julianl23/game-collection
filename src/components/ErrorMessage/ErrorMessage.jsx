import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Message = styled.div`
  color: ${({ theme }) => theme.pinkRed};
  font-size: 1em;
  font-weight: 500;
`;

const ErrorMessage = ({ className, message }) => (
  <Message className={className}>{message}</Message>
);

ErrorMessage.propTypes = {
  className: PropTypes.string,
  message: PropTypes.string.isRequired,
};

ErrorMessage.defaultProps = {
  className: "",
};

export default ErrorMessage;
