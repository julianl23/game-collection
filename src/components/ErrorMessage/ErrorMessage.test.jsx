import React from "react";
import TestRenderer from "react-test-renderer";
import ErrorMessage from "./ErrorMessage";

describe("ErrorMessage", () => {
  it("renders an error message", () => {
    const instance = TestRenderer.create(
      <ErrorMessage message="Hello World" />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
