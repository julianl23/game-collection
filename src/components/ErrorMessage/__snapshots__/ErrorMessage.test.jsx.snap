// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders an error message 1`] = `
.c0 {
  font-size: 1em;
  font-weight: 500;
}

<div
  className="c0"
>
  Hello World
</div>
`;
